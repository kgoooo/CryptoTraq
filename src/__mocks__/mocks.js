export const mockStyles = {
	mockAnim: {
		boxShadow: '0 0 3px 6px blue'
	},
	mockBtn:{
		"background": "transparent",
		"border": "2px solid #00FFF5",
		"borderRadius": "8px",
		"color": "#00FFF5",
		"fontSize": "inherit",
		"fontWeight": "300",
		"padding": ".5rem 2.5rem",
		"transition": "all ease-in-out .3s"
	}
};

export const mockMethod = jest.fn();

export const mockCryptoList = [
	{name: "Bitcoin", code: "BTC"},
	{name: "Ethereum", code: "ETH"},
	{name: "Doge", code: "DOGE"}
];

export const mockIntlList = [
	{code: "USD"},
	{code: "CAD"},
	{code: "EUR"},
	{code: "GBP"}
];

export const mockCryptoData = { data: [{
		is_crypto: true,
		name: "crypto1",
		code: "c1"
	}, {
		is_crypto: true,
		name: "crypto2",
		code: "c2"
	}, {
		is_crypto: false,
		name: "fakemoney1",
		code: "fk1"
	}, {
		is_crypto: false,
		name: "fakemoney2",
		code: "fk2"
	}]}

export const mockGraphData = [
	{
		zuluTime: "2018-08-16T19:43:20.221129Z",
		unixTime: 1534439534,
		rate: 623,
	},{
		zuluTime: "2018-08-16T16:42:19.697216Z",
		unixTime: 1534441338,
		rate: 627,
	},{
		zuluTime: "2018-08-16T17:12:14.580348Z",
		unixTime: 1534443072,
		rate: 629,
	},{
		zuluTime: "2018-08-16T17:42:18.458736Z",
		unixTime: 1534444934,
		rate: 632,
	},
];

export const mockGraphDataDesktop = {data: [
		{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}
	]};

export const mockGraphDataMobile = {
	data: [
		{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}
	]
};
