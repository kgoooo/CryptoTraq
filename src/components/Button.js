/*
		This is the new button component that I made as a result of adding Radium.  Here I can keep my code 'DRY'
		by having just one Button component.  I wanted to remove CSS files completely by using Radium for this app
		and found it to be impossible to not repeat code unless the app is truly reusable with its components.
 */

import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import RadiumVars from '../RadiumVariables';

/*  Stateless Button components, the button content and methods get passed through props now.  Before each
		button was configured manually.
*/
/*
		Added the animation style for the button, passed in through props.
 */
const Button = (props) => {
	return (
		<button
			onClick={props.onClick}
			style={[styles.btn__main, props.buttonAnimation]}
		>{props.buttonContent}</button>
	)
};

Button.propTypes = {
	onClick: PropTypes.func.isRequired,
	buttonContent: PropTypes.string.isRequired,
	buttonAnimation: PropTypes.object
};

/*  Button styles for Radium defined here and added below to the component on the export statement. */
const styles = {
	btn__main: {
		background: 'transparent',
		borderRadius: '8px',
		border: `2px solid ${RadiumVars.color.colorPrimaryLightBlue}`,
		color: RadiumVars.color.colorPrimaryLightBlue,
		fontSize: 'inherit',
		fontWeight: '300',
		padding: '.5rem 2.5rem',
		transition: 'all ease-in-out .3s',

		':hover': {
			boxShadow: `0 0 3rem ${RadiumVars.color.colorPrimaryBlue}`
		}
	}
};

export default Radium(Button);
