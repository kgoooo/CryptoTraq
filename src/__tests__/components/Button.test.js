import React from 'react';
import { shallow } from 'enzyme';
import { mockStyles, mockMethod} from '../../__mocks__/mocks';
import Button from '../../components/Button'

describe('<Button />', ()=> {

	it('Snapshot: should render Button correctly', () => {
		const wrapper = shallow(<Button/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should add onClick method when props passed in', () => {
		const wrapper = shallow(<Button onClick={mockMethod}/>);
		expect(wrapper.props().onClick).toEqual(mockMethod);
	});

	it('Should add animation style when props passed in', () => {
		const wrapper = shallow(<Button style={[mockStyles.mockBtn, mockStyles.mockAnim]}/>);
		expect(wrapper.instance().props).toMatchObject({
		"style": [
			{"background": "transparent",
				"border": "2px solid #00FFF5",
				"borderRadius": "8px",
				"color": "#00FFF5",
				"fontSize": "inherit",
				"fontWeight": "300",
				"padding": ".5rem 2.5rem",
				"transition": "all ease-in-out .3s"},
			{
				"boxShadow": "0 0 3px 6px blue"
			}
			]})
	});

	it('should add text correctly if passed buttonContent prop', () => {
		const wrapper = shallow(<Button buttonContent={"I'm a button"}/>);
		expect(wrapper.text()).toBe("I'm a button");
	});

	it('should call mockMethod on click', () => {
		const wrapper = shallow(<Button onClick={mockMethod} />);
		wrapper.simulate('click');
		expect(mockMethod).toHaveBeenCalled();
	})
});