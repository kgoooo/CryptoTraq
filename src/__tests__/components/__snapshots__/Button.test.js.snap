// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshot: should render Button correctly 1`] = `
<button
  data-radium={true}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "transparent",
      "border": "2px solid #00FFF5",
      "borderRadius": "8px",
      "color": "#00FFF5",
      "fontSize": "inherit",
      "fontWeight": "300",
      "padding": ".5rem 2.5rem",
      "transition": "all ease-in-out .3s",
    }
  }
/>
`;
